toolkit/tools/pkgworker/pkgworker.go

```diff
@@ -238,6 +239,7 @@ func buildSRPMInChroot(chrootDir, rpmDirPath, toolchainDirPath, workerTar, srpmF
 	case <-time.After(timeout):
 		logger.Log.Errorf("Timeout after %v: killing all processes in chroot...", timeout)
 		shell.PermanentlyStopAllChildProcesses(unix.SIGKILL)
+		time.Sleep(5 * time.Second) // If we don't wait, the chroot mounts may still be busy
```

Comment:
This is a very unreliable way of doing this. Please try to find an alternative signal to wait on.

Reply:
This is an edge-case of an edge-case type thing, but I'll see if there is a simple way to handle this better.

I'm going to say we just keep this hack if there isn't an easy fix. This failure case is very uncommon, and worst case just leaves a folder in place on the system.

I'm going to look at the gpg key service handing code, it might be failing since we call `PermanentlyStopAllChildProcesses()` and this code wasn't considered when we added the gpg stuff since its so rare.

Reply:
Ok, figured out the underlying issue here.